### PR TITLE
Update formatting in the release procedure section of maintaining.rst

### LINKED ISSUE #50824: DOC: Use rst instead of markdown in release instructions
Looks like when I moved the release procedure to the pandas docs I didn't realize I had the notes in markdown, and the docs are in rst. The formatting is all broken because of this: https://pandas.pydata.org/docs/dev/development/maintaining.html#release-process

We should replace the mardown with rst, so sphinx can render things correctly.

### PR DESCRIPTION
**Pull request by tpackard1**
- [ ] closes #50824 

@datapythonista what do you think about these updates so far?

### CODE REVIEW

File: doc/source/development/maintaining.rst [hunk 1]
```diff
@@ -432,29 +432,29 @@ which will be triggered when the tag is pushed.
...
 
-6. Packages for supported versions in PyPI are built in the
-   [MacPython repo](https://github.com/MacPython/pandas-wheels).
+#. Packages for supported versions in PyPI are built in the
+   `MacPython repo <https://github.com/MacPython/pandas-wheels>`_.
    Open a PR updating the build commit to the released version, and merge it once the
-   CI is green.
-
+   CI is green. Update it by typing::
+
```
Comment (datapythonista) [inline doc/source/development/maintaining.rst]: I think a line with spaces will make the CI fail.

**Review by datapythonista — COMMENTED** [review]
Looks good, feel free to open a PR once you're done.

**Comment by tpackard1** [comment]
@datapythonista not sure how to interpret the Doc Build and Upload / Doc Build and Upload (pull_request) failing particularly where and what went wrong. Any ideas would be greatly appreciated.

**Comment by datapythonista** [comment]
You opened this branch against your own fork. Can you open it to the main pandas repo, and I'll have a look. Thanks!

**Comment by tpackard1** [comment]
After making sure that my master branch is up to date with main pandas repo. Should I do this by doing the following?
```bash
git checkout format-release-process
git rebase master
```

**Pull request closed without merging**